node/src/actors/chain_manager/mod.rs

```diff
@@ -651,7 +653,7 @@ fn update_reputation(
 
     // Update active reputation set
     // Add block miner pkh to active identities
-    rep_eng.ars.push_activity(revealers.chain(vec![miner_pkh]));
+    rep_eng.update_ars(block_epoch, revealers.chain(vec![miner_pkh]));
```

Comment:
Since the `ars` is only updated here, it can lead to incorrect results. Consider the case when we are in epoch 100 and there are 40 epochs with no block and the `ars` has a maximum size of 40. The set of active identities should be empty, but it won't be updated until after processing the block for epoch 140. Therefore in the reputation validations for block 140 we will count identities which have participated in epochs 60-99 as active, when clearly they weren't active because in epochs 100-139 there were no blocks.

Reply:
It is also updated in ChainManager/handler.rs 138, in case of an empty epoch